src/main/java/ch/fhnw/algd1/converters/binary/BinConverter.java

```diff
@@ -4,12 +4,83 @@ public class BinConverter {
 	public static String toString(int x) {
 		// TODO expect x to be in range [-128, 127], return string with 8 binary
 		// digits representing x in 2-complement
-		return "00000000";
+		String result = "";
+		boolean positive = x>=0;
+		if(positive){
+			while(x>0){
+				if(x%2 == 0){
+					result = "0"+result;
+				} else {
+					result = "1"+result;
+				}
+				x = x/2;
+			}
+			int count = 8-result.length();
+			for(int i = 0; i < count; i++ ){
+				result = "0"+result;
+			}
+		} else{
+			x = Math.abs(x);
+			String intermediateResult = "";
+			while(x>0){
+				if(x%2 == 0){
+					intermediateResult = "0"+intermediateResult;
+				} else {
+					intermediateResult = "1"+intermediateResult;
+				}
+				x = x/2;
+			}
+			int count = 8-intermediateResult.length();
+			for(int i = 0; i < count; i++ ){
+				intermediateResult = "0"+intermediateResult;
+			}
+
+			for(int i = 0; i< intermediateResult.length(); i++){
+				if(intermediateResult.charAt(i) == '1'){
+					result += 0;
+				} else{
+					result += 1;
+				}
+			}
+
+
+			for(int i = result.length()-1; i >= 0; i--){
+				int memory = 0;
+				if(result.charAt(i)=='0'){
+					result = replaceChar(result, '1',i);
+					return result;
+				} else {
+					result = replaceChar(result, '0',i);
+				}
+			}
```

Comment:
Das liesse sich jetzt gut mit der aktuell neu gelernten Idee der "Sequenziellen Suche" verbinden: Suche den höchsten Index, bei dem der Wert 0 steht und setze alle Werte mit höherem Index auf 0. In diesem Fall stoppt die Suche immer, weil ja das Bit 0 auf jeden Fall 0 ist.
```suggestion
            int i = result.length() - 1;
            while (result.charAt(i) != '0') {
                result = replaceChar(result, '0', i);
                i--;
            }
            result = replaceChar(result, '1', i);
```